File: 01-week-1--starter-algorithms/01-day-2--find-first-duplicate/javascript/find_first_duplicate.js
function findFirstDuplicate(arr) {

  console.log(arr)
  for (i = 0; i < arr.length ; i ++){
      if (arr[i] == arr[i++]){
        
      }
    }
    
  }

  
  

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 3");
  console.log("=>", findFirstDuplicate([2, 1, 3, 3, 2]));

  console.log("");

  console.log("Expecting: -1");
  console.log("=>", findFirstDuplicate([1, 2, 3, 4]));

  console.log("");

  console.log("Expecting: 1");
  console.log("=>", findFirstDuplicate([1, 1 ,2, 3, 4]));

  console.log("");

  console.log("Expecting: -1");
  console.log("=>", findFirstDuplicate([]));
}

module.exports = findFirstDuplicate;

// pseudocode:
// compare each element of array, to all elements of array
// nested for loop

File: 01-week-1--starter-algorithms/01-day-2--find-first-duplicate/ruby/find_first_duplicate.rb
require 'set'

def find_first_duplicate(arr)

  unique_numbers = Set.new

  arr.each do |value|
    return value if unique_numbers.include?(value)
    unique_numbers.add(value)
  end

  -1

end


if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 3"
  puts "=>", find_first_duplicate([2, 1, 3, 3, 2])

  puts

  puts "Expecting: -1"
  puts "=>", find_first_duplicate([1, 2, 3, 4])

  puts "Expecting: -1"
  puts "=>", find_first_duplicate([])
end

# pseudocode:
# does ruby have same Set functionality as JS?